src/Entity/ContentItem.php

```diff
@@ -35,22 +35,33 @@ public function getShortType($infix = '-')
      *
      * @param ContentItemInterface $from
      * @param ContentItemInterface $to
-     * @return ContentItemInterface
+     * @return array
      */
     public static function convert(ContentItemInterface $from, ContentItemInterface $to)
     {
-        $reflectionFrom = new \ReflectionClass($from);
-        $reflectionTo = new \ReflectionClass($to);
-
-        foreach ($reflectionFrom->getProperties() as $property) {
-            $property->setAccessible(true);
-            $method = 'set' . ucfirst($property->getName());
-            if ($reflectionTo->hasMethod($method)) {
-                $to->$method($property->getValue($from));
-            }
+        $properties = [];
+        try {
+            $reflectionFrom = new \ReflectionClass($from);
+            $reflectionTo = new \ReflectionClass($to);
+
+            do {
+                $convertToType = [];
+                /* @var $p \ReflectionProperty */
+                foreach ($reflectionFrom->getProperties() as $property) {
+                    $property->setAccessible(true);
+                    $method = 'set' . ucfirst($property->getName());
+
+                    if ($reflectionTo->hasMethod($method)) {
+                        $to->$method($property->getValue($from));
+                        $convertToType[$method] = $to->$method($property->getValue($from));
+                    }
```

Comment:
Deze code was er natuurlijk al en is prima, maar mocht je het nog net 1 tandje scherper neer willen zetten:

```suggestion
                    $method = 'set' . ucfirst($property->getName());
                    if (!$reflectionTo->hasMethod($method)) {
                        continue;
                    }

                    $property->setAccessible(true);
                    $to->$method($property->getValue($from));
                    $convertToType[$method] = $to->$method($property->getValue($from));
```

Reply:
Aangepast